### PR TITLE
Updating the migrating doc

### PR DESCRIPTION
**Pull request by vincentpierre**

### CODE REVIEW

File: docs/Migrating.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Migrating
 
+## Migrating from ML-Agents toolkit v0.4 to v0.5
+
+### Unity API
+ * `unity-environment` has been renamed `MLAgentsSDK`.
+
+### Python API
+ * The `python` folder has been reorganized. We removed the `python` folder and replaced it with `ml-agents`. The folder `ml-agents` now contains the packages `mlagents.env` to interact directly with Unity environments and `mlagents.trainers` to train environments.
```
Comment (awjuliani) [inline docs/Migrating.md]: Here you say that we both reorganized the `python` folder and removed it. Perhaps it is better to say at the beginning of this section that we reorganized the entire repo, and then list the changes.

File: docs/Migrating.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
 # Migrating
 
+## Migrating from ML-Agents toolkit v0.4 to v0.5
+
+### Important
+ * We reorganized the entire repository
```
Comment (mmattar) [inline docs/Migrating.md]: not sure we need this

File: docs/Migrating.md [hunk 3]
```diff
@@ -1,5 +1,17 @@
...
+ * `unity-environment` has been renamed `MLAgentsSDK`.
```
Comment (mmattar) [inline docs/Migrating.md]: what is `unity-environment`? Folder, package, file?

File: docs/Migrating.md [hunk 4]
```diff
@@ -1,5 +1,17 @@
...
+ *  The `python` folder has been reorganized. We removed the `python` folder and replaced it with `ml-agents`. The folder `ml-agents` now contains the packages `mlagents.env` to interact directly with Unity environments and `mlagents.trainers` to train environments.
```
Comment (mmattar) [inline docs/Migrating.md]: Change too ....

The `python` folder has been renamed to `ml-agents`. It not contains two packages, `mlagents.env` and `mlagents.trainers`. `mlagents.env` can be used to interact directly with a Unity environment, while `mlagents.trainers` contains the classes for training agents.

File: docs/Migrating.md [hunk 5]
```diff
@@ -1,5 +1,17 @@
...
+
+### Python API
+ * The command `python3 learn.py` has been replaced with `mlagents-learn`.
```
Comment (mmattar) [inline docs/Migrating.md]: what is `mlagents-learn`? Python script?

File: docs/Migrating.md [hunk 6]
```diff
@@ -1,5 +1,17 @@
...
+ * It is now required to specify the path to the `trainer-config.yaml` file when running `mlagents-learn`.
```
Comment (mmattar) [inline docs/Migrating.md]: Do we specify path to the file or the full path? In other words, does the YML file always need to be called `trainer-config.yaml`?

File: docs/Migrating.md [hunk 7]
```diff
@@ -1,5 +1,34 @@
...
+
+### Unity API
+
+* Discrete Actions now have branches. You can now specify concurrent discrete
+  actions. You will need to update the Brain Parameters in the Brain Inspector
+  in all your environments.
+
+### Python API
+
+* In order to run a training session, you must now use the command
```
Comment (dericp) [inline docs/Migrating.md]: I think we should let them know that the optional executable environment is passed by the `--env` argument. Maybe we could also link to the doc specifying how to use `mlagents-learn`?

File: docs/Migrating.md [hunk 8]
```diff
@@ -1,5 +1,34 @@
...
+* The `python` folder has been renamed to `ml-agents`. It not contains two
+  packages, `mlagents.env` and `mlagents.trainers`. `mlagents.env` can be used
+  to interact directly with a Unity environment, while `mlagents.trainers`
+  contains the classes for training agents.
+
+### Unity API
+
+* Discrete Actions now have branches. You can now specify concurrent discrete
+  actions. You will need to update the Brain Parameters in the Brain Inspector
+  in all your environments.
```
Comment (xiaomaogy) [inline docs/Migrating.md]: Maybe we should have a visual demonstration on how they should do the switch for the brain parameters?

File: docs/Migrating.md [hunk 9]
```diff
@@ -1,5 +1,34 @@
...
+
+### Python API
+
+* In order to run a training session, you must now use the command
+  `mlagents-learn` instead of `python3 learn.py` after installing the `mlagents`
+  packages.
+* It is now required to specify the path to the yaml trainer configuration file
+  when running `mlagents-learn`. For example, see
+  [trainer_config.yaml](../config/trainer_config.yaml).
```
Comment (xiaomaogy) [inline docs/Migrating.md]: Why did we make this change? Do we have an example of the previous command vs current command?

Comment (dericp) [reply docs/Migrating.md]: The reasoning for this change was that no matter what environment we're training, we need a trainer configuration file. Although we have sensible defaults for the environments we ship in `trainer_config.yaml`, it doesn't makes sense to have users add their custom configurations to this file or potentially get confused when default configurations are getting loaded.

By forcing the user to specify the configuration file, it makes it explicit where the hyper-parameters are coming from.

File: docs/Migrating.md (hunk 8 above)
Comment (vincentpierre) [reply docs/Migrating.md]: I don't think this is needed

File: docs/Migrating.md (hunk 7 above)
Comment (vincentpierre) [reply docs/Migrating.md]: added a line for that

File: docs/Migrating.md (hunk 8 above)
Comment (xiaomaogy) [reply docs/Migrating.md]: ok = =

**Pull request merged**